packages/viewer/src/components/viewer-transform-widget/util.ts

```diff
@@ -99,22 +100,40 @@ export function computeInputTransform(
 
   switch (identifier) {
     case 'x-translate':
-      return Matrix4.makeTranslation(Vector3.create(position(), 0, 0));
+      return appliedToCurrent(
+        current,
+        Matrix4.makeTranslation(Vector3.create(position(), 0, 0))
+      );
     case 'y-translate':
-      return Matrix4.makeTranslation(Vector3.create(0, position(), 0));
+      return appliedToCurrent(
+        current,
+        Matrix4.makeTranslation(Vector3.create(0, position(), 0))
+      );
     case 'z-translate':
-      return Matrix4.makeTranslation(Vector3.create(0, 0, position()));
+      return appliedToCurrent(
+        current,
+        Matrix4.makeTranslation(Vector3.create(0, 0, position()))
+      );
     case 'x-rotate':
-      return Matrix4.makeRotation(
-        Quaternion.fromAxisAngle(Vector3.left(), rotation())
+      return appliedToCurrent(
+        current,
+        Matrix4.makeRotation(
+          Quaternion.fromAxisAngle(Vector3.left(), rotation())
+        )
       );
     case 'y-rotate':
-      return Matrix4.makeRotation(
-        Quaternion.fromAxisAngle(Vector3.down(), rotation())
+      return appliedToCurrent(
+        current,
+        Matrix4.makeRotation(
+          Quaternion.fromAxisAngle(Vector3.down(), rotation())
+        )
       );
     case 'z-rotate':
-      return Matrix4.makeRotation(
-        Quaternion.fromAxisAngle(Vector3.forward(), rotation())
+      return appliedToCurrent(
+        current,
+        Matrix4.makeRotation(
+          Quaternion.fromAxisAngle(Vector3.forward(), rotation())
+        )
```

Comment:
This `appliedToCurrent` behavior fixes an issue where starting rotations on each axis would result in incorrect rotation calculation. This then caused the input to jump from something like `270` to `277` when pressing enter